test/e2e-tests.sh

```diff
@@ -85,6 +85,25 @@ $(dirname $0)/upload-test-images.sh e2e || fail_test "Error uploading test image
 # Setup resources common to all eventing tests
 setup_events_test_resources|| fail_test "Error setting up test resources"
 
-go_test_e2e ./test/e2e || fail_test
+go_test_e2e ./test/e2e 
+exit_result=$?
+if [ ${exit_result} -ne 0 ]; then
+# Collecting logs from all knative's eventing pods
+  for namespace in "knative-eventing" "e2etestfn3"; do
+    for pod in $(kubectl get pod -n $namespace | grep Running | awk '{print $1}' ); do
+      echo "========== Collecting logs for pod: "${pod}" in $namespace ================="
+      for container in $(kubectl get pod "${pod}" -n $namespace -ojsonpath='{.spec.containers[*].name}'); do
+        echo "----------------------------------------------------------"
+        echo "Container: "${container}
```

Comment:
My preference is to include all the pieces on one line (preferably in an easily searchable way). Something like:

```shell
echo "Namespace, Pod, Container: ${namespace}, ${pod}, ${container}"
```

Reply:
done